c/core.go

```diff
@@ -75,15 +86,15 @@ func NewWithNotifyStart(
 	}
 	spec.name = name
 
-	if start == nil {
+	if childMain == nil {
 		panic("Child cannot have empty start function")
```

Comment:
Ideally, this would be a compilation error, in an ideal world, it would not be possible to create a `ChildSpec` without a `start` function. The closes I could do is to fail hard on invalid `ChildSpec`, this should never happen if clients use the `New` function. Maybe adding some documentation of this invariant in `ChildSpec` can help?

Reply:
No, no, I was just remembering the scar related to the possibility of remote resources being left hanging because of how abrupt `panic` _might_ be. I need to do more research on this.